src/topology/locally_constant/algebra.lean

```diff
@@ -110,4 +110,27 @@ instance [ring Y] : ring (locally_constant X Y) :=
 instance [comm_ring Y] : comm_ring (locally_constant X Y) :=
 { .. locally_constant.comm_semiring, .. locally_constant.ring }
 
+variables (M : Type*) [monoid M] (A : Type*) [semiring A]
+
+instance : has_scalar M (locally_constant X M) :=
```

Comment:
I think it would be worth adding some lemmas about this action below:
```
@[simp] lemma coe_smul (f : locally_constant X A) (a : A) : ⇑(a • f) = a • (f : X → A) := rfl

lemma smul_apply (f : locally_constant X A) (a : A) (x : X) : (a • f) x = a • (f x) := rfl
```

Reply:
I'd probably flip the `f` and `m` arguments there to match the order in the lemma statement